R/backend-mssql.R

```diff
@@ -144,23 +144,35 @@ mssql_temp_name <- function(name, temporary){
                             analyze = TRUE, ...) {
   NextMethod(
     table = mssql_temp_name(table, temporary),
+    types = types,
+    values = values,
     temporary = FALSE
   )
 }
 
 #' @export
 `db_save_query.Microsoft SQL Server` <- function(con, sql, name,
                                                  temporary = TRUE, ...){
-  NextMethod(
-    name = mssql_temp_name(name, temporary),
-    temporary = FALSE
+  name <- mssql_temp_name(name, temporary)
+
+  # Different syntax for MSSQL: https://stackoverflow.com/q/16683758/946850
+  tt_sql <- build_sql(
+    "SELECT * ",
+    "INTO ", as.sql(name), " ",
+    "FROM (", sql, ") AS temp",
+    con = con
   )
+  dbExecute(con, tt_sql)
```

Comment:
Could we test this without an activate MS SQL backend by mocking `dbExecute()` to just return the sql, and then using `expect_known_output()`?

Reply:
Perhaps with an approach similar to dbi-log. But then we still need to double-check the generated SQL.

Would you like me to open a new issue to discuss?

Reply:
No, I just want a simple way to test this code without adding additional complexity or new dependencies.

Reply:
Filed an issue: #284. Suggesting to rely on user testing for this PR, discuss/implement #284 for the v1.4.0 release and add some tests too.